src/Symfony/Component/Translation/Translator.php

```diff
@@ -459,8 +502,13 @@ private function loadFallbackCatalogues($locale)
         }
     }
 
+    /**
+     * @deprecated since version 2.8, to be removed in 3.0. Rely on FallbackTranslatorBag instead.
+     */
     protected function computeFallbackLocales($locale)
     {
+        trigger_error('The '.__METHOD__.' method is deprecated since version 2.8 and will be removed in 3.0. Rely on FallbackTranslatorBag instead.', E_USER_DEPRECATED);
```

Comment:
Two things –

1. Is that the right way to deprecate `protected` functions? I mean, people might override this method without calling the parent class and never get the deprecation notice. (@stof?)

2. How can people provide their own implementation for this? Currently this is possible. That also applies to other protected methods as well.